### ISSUE TITLE
[META] Replace bytestrings across all fetchers with unicode strings

### ISSUE THREAD

**Issue by kchawla-pi** [nilearn/nilearn#2263]
Most atlases & many fetchers currently use byte-strings in their labels and descriptions etc. This is suboptimal and should be changed to regular strings.

3 examples of this are #1635 , #1181  and `fetch_haxby` .

Question, this will be a backwards incompatible change. Does that matter for fetchers used didactic training?
Will it break anything with the atlases, which might be used in the code?

Everyone's input here is appreciated.

**Comment by kchawla-pi** [comment]
The fetchers & atlases won't all be changed in one PR, probably each fetcher & atlas will have its own PR. SO this is a Meta bug.

**Comment by kchawla-pi** [comment]
I will heed Alexandre's advice https://github.com/nilearn/nilearn/issues/1181#issuecomment-252619872

**Comment by kchawla-pi** [comment]
I'll start with `fetch_haxby`

**Comment by jeromedockes** [comment]
> Question, this will be a backwards incompatible change.

indeed. are you planning a deprecation cycle? also, all the changes should ideally happen in the same release
